plonky2/src/plonk/circuit_data.rs

```diff
@@ -139,6 +139,18 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
     {
         compressed_proof_with_pis.verify(&self.verifier_only, &self.common)
     }
+
+    pub fn verifier_data(self) -> VerifierCircuitData<F, C, D> {
```

Comment:
If we add this in the public API I think we should have a similar `prover_data` method.

Let's also update `build_prover` and `build_verifier` to use them.

Reply:
Good idea, done